File: docs/code-examples/asset3d_simple.cpp
// Log a a simple 3D asset.

#include <rerun.hpp>

#include <filesystem>
#include <iostream>
#include <string>
#include <vector>

int main(int argc, char* argv[]) {
    std::vector<std::string> args(argv, argv + argc);

    if (args.size() < 2) {
        std::cerr << "Usage: " << args[0] << " <path_to_asset.[gltf|glb|obj]>" << std::endl;
        return 1;
    }

    std::string path = args[1];

    auto rec = rerun::RecordingStream("rerun_example_asset3d_simple");
    rec.connect().throw_on_failure();

    rec.log_timeless("world", rerun::ViewCoordinates::RIGHT_HAND_Z_UP); // Set an up-axis
    rec.log("world/asset", rerun::Asset3D::from_file(path).value_or_throw());
}

File: docs/code-examples/asset3d_simple.py
"""Log a a simple 3D asset."""
import sys

import rerun as rr
import rerun.experimental as rr2

if len(sys.argv) < 2:
    print(f"Usage: {sys.argv[0]} <path_to_asset.[gltf|glb]>")
    sys.exit(1)

rr.init("rerun_example_asset3d_simple", spawn=True)

rr2.log("world", rr2.ViewCoordinates.RIGHT_HAND_Z_UP, timeless=True)  # Set an up-axis
rr2.log("world/asset", rr2.Asset3D.from_file(sys.argv[1]))
